README.md

```diff
@@ -0,0 +1,169 @@
+For a more detailed description of the app to the Project section.
+
+## Installation
+
+1. **Prerequisites:** The project needs [Node](https://nodejs.org) ^10.13.0 and [Yarn](https://yarnpkg.com) ^1.12.3 to as a dependency manager.
+2. Download or clone the project to a local folder
+3. While in the project folder run `yarn install`
+4. When all dependencies are installed, run `yarn start` to start the dev server
+
+### The Project
+
+### Features
+
+-   Show a list of dogs retrieved from [DogAPI](https://dog.ceo/dog-api/)
+-   Filter the list by dog breeds through a text input
+-   Show an image of each breed retrieved from [DogAPI](https://dog.ceo/dog-api/)
+-   Each dog breed name appears on a tooltip when you hover over the image
+-   Each dog has its own page with details retrieved from [MediaWiki](https://www.mediawiki.org/wiki/API:Main_page)
+-   Submit a dog picture through a form
+-   Each dog card is only rendered when the component is in the viewport, by using a lazy load component
+-   Each dog image has a preload of the image and displays a placeholder if there were loading errors
+
+### Architecture
+
+-   The project uses React for UI rendering
+-   Redux is used as a state manager for sharing state between non immediate parent/child components
+-   A `<Router />` component was created from scratch based on the [react-router](https://github.com/ReactTraining/react-router) implementation
+-   The router components use **react context** as a state manager instead of redux as the router needed the easiest way to pass data between components
+
+#### State Management
+
+As the app kept getting more features passing data between components became more complicated, as a way to simplify the state passed between components, Redux was implemented. This can be seen in the `<SubmitDog />` and `<AllDogs />` components which share the same data.
+
+As Redux's actions don't wait asynchronous functions, the [redux-thunk](https://github.com/reduxjs/redux-thunk) middleware was used for resolving this behavior.
+
+In this app, Redux handles the fetching and sending of data from the APIs as well as managing this information in the store. It also stores and removes messages from the message list that `<MessageHandler />` uses.
+
+### Key parts
+
+#### `<App />`
+
+The root component of the App.
+Connects with the Redux store and actions to be passed as props.
+On mount, the component fetches the dog's list from local storage if present or from the API to be set on store.
+
+#### Router
+
+The `<Router />` component was coded from scratch based on the [react-router](https://github.com/ReactTraining/react-router) library. It has four main components: `<Router />`, `<Route />`, `<Redirect />` and `<Otherwise/ >`. There's also a `<Link />` component to assist with navigation.
+Uses **React Context** to handle the state and pass it to components that may need it.
+
+[Read more about the router](./components.md#router)
+
+#### ShowDogs Components
```

Comment:
The component is DogList but the folder with all the components regarding the showing of dogs is called ShowDogs, thats why I don't put it as <ShowDogs />. It would be better to just title the section as DogList Components?